### PR TITLE
Create ratio_location.yaml

### PR DESCRIPTION
**Pull request by ar-ibrahim**
- issue #272 
- Rough (very) draft of 'ratio' pattern, including proposed name of pattern.
- To discuss in monthly OBA meeting (July 21st 2023)

### CODE REVIEW

File: src/patterns/dosdp-patterns/ratio_location.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+pattern_name: ratio_location
+pattern_iri: http://purl.obolibrary.org/obo/oba/patterns/ratio_location.yaml
+
+description: '
```
Comment (rays22) [inline src/patterns/dosdp-patterns/ratio_location.yaml:4]:
```suggestion
description: 'Use this pattern for traits that that depend on the proportions, ratios, or inverse ratios of two measurable attributes. Although calculating a ratio is not commutative, the trait category as defined here is indifferent about the direction of the ratio between the two quantifiable attributes, i.e. A:B and B:A are assumed to represent reciprocal aspects of the same abstract biological attribute that share the same genetic determinants. For example, the sodium to potassium ratio is assumed to have the same genetic determinants in an organism than the potassium to sodium ratio, and the resulting reciprocal values are not independent from each other.'
```

File: src/patterns/dosdp-patterns/ratio_location.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
```
Comment (rays22) [inline src/patterns/dosdp-patterns/ratio_location.yaml:5]:
```suggestion
```

File: src/patterns/dosdp-patterns/ratio_location.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  Basic pattern that captures the proportion of an entity in respect to a related entity, in some location.'
```
Comment (rays22) [inline src/patterns/dosdp-patterns/ratio_location.yaml:6]:
```suggestion
```

**Review by rays22 — DISMISSED** [review]
We might want to change the name of the pattern, because it does not specify the location explicitly. The components attribute1 and attribute2 should specify and imply the location of the ratio-attribute defined here.

File: src/patterns/dosdp-patterns/ratio_location.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    vars:
+      - location
+      - entity
+    annotations:
+      - annotationProperty: xref
+        text: "AUTO:patterns/patterns/ratio_location"
+
+
+equivalentTo:
+  text:
```
[the remark this answers is not in the document]
Comment (matentzn) [reply src/patterns/dosdp-patterns/ratio_location.yaml:62]: Try to design this in a way that allows grouping of abnormal rations from both sides.

[the remark this answers is not in the document]
Comment (matentzn) [reply src/patterns/dosdp-patterns/ratio_location.yaml:62]: 1. Take an abnormal ratio of X term in HPO and create a pattern for that
2. Then ensure that the OBA pattern captures all four sides of the ratio: increased a to b, increased b to a, decrease a to b, decreased b to a (the reasoner should infer the former to be a sublcass of the OBA one.)
3.

**Comment by ar-ibrahim** [comment]
I have consulted @rays22 about the recurring QC warnings and as per my understanding this has been an outstanding issue that does not prevent merging the PR should the suggested changes be approved.

**Pull request merged**